app/controllers/api/v1/search_controller.rb

```diff
@@ -503,9 +503,25 @@ def set_search_facets_and_filters
             facet = SearchFacet.find_by(identifier: facet_id)
             if facet.present?
               matching_filters = []
-              facet.filters.each do |filter|
-                if filter_values.include?(filter[:id])
-                  matching_filters << filter
+              if facet.is_numeric?
+                # first filter value is range, e.g. 20-40, second value is unit, e.g. years
+                min_value, max_value = self.class.split_query_param_on_delim(parameter: filter_values.first, delimiter: '-').map(&:to_f)
+                requested_unit = filter_values.last
```

Comment:
We haven't defined URL parameters syntax for unit-less numeric facets.  But you could imagine `number_of_reads:150-500`.  Note the lack of units.

In that case, `filter_values.last` might not work.  Something to consider for later, at least.

Reply:
That was the thought I had yesterday evening.  I need logic to specifically check for units by looking in the `SearchFacet::TIME_UNITS` constant.